src/layout/layout.js

```diff
@@ -99,44 +98,47 @@ function _layout() {
     return layout;
 }
 
-function layoutAdapter(name, value) {
-    // Quick bodge to fix #568
-    if (this.node() == null) {
-        return this;
+function layoutSelection(name, value) {
+    var argsLength = arguments.length;
+
+    // For layout(string), return the lyout value for the first node
```

Comment:
No it doesn't - it follows the same pattern as D3's `attr`:

https://github.com/mbostock/d3/blob/master/src/selection/attr.js#L9

Just to be sure, I added an additional test